code/modules/projectiles/projectile.dm

```diff
@@ -325,9 +325,16 @@
 
 	if(ismob(firer))
 		log_combat(firer, L, "shot", src, reagent_note)
-	else
-		L.log_message("has been shot by [firer] with [src]", LOG_ATTACK, color="orange")
+		return BULLET_ACT_HIT
+
+	if(ismecha(firer))
+		var/obj/vehicle/sealed/mecha/chassis = firer
+		var/mob/occupant = chassis.occupants[1]
```

Comment:
mecha can have multiple people in it controlling different systems. (ie one is a pilot, one controls weapons, ect)

this needs to be taken up to isvehicle() regardless. 

use `vehicle.return_controllers_with_flag(VEHICLE_CONTROL_EQUIPMENT)` for a list of mobs that have equipment (weapons) access.